### ISSUE TITLE
Migrate from Current Plugin to This?

### ISSUE THREAD

**Issue by mrjarbenne** [hwdsbcommons/reveal-js-presentations#11]
@ellatrix has created a cool variation on this that leverages the new Editor. This might be a good place to focus our development efforts, given the momentum it has right now. See repo here: https://github.com/WordPress/slides

**Comment by r-a-y** [comment]
So do we want to create a migration script or something along those
lines for existing users of reveal-js-presentations?

If so, I'd need to see what tthe differences are between the two plugins
to see how they both store their data.

**Comment by mrjarbenne** [comment]
No, I think we can just switch it over. All the slides are time sensitive, and set to unpublish after seven days. The work might be in getting the other plugins that we rely on to make the service work compatible with the new editor. https://github.com/humanmade/unpublish would be a good example of something that's holding us back. Maybe we can find an Unpublish solution.

**Comment by r-a-y** [comment]
I looked at the tweaks we made to Unpublish via the "Announcements Unpublish Tweaks" plugins and just added support for the `presentation` post type, which the Slides plugin uses.

I have no idea if that will work or not, but let me know.

**Comment by mrjarbenne** [comment]
Upon further digging I don't think it will work for our application. Individual slides need to be able to have expiry dates on them. Given that the Presentation is the Post Type, and the slides are all just blocks within the post type, editing the content of the presentation raises permissions issues. Our current logic, where each slide is a post, allows the flexibility and permissions we need. 

I wonder though, if the work being done to add a Slides block could be forked for our application once it's fully formed. Currently our plugin uses the old editor by turning the REST filter to false, but eventually we will probably need to port it over to the new editor. Having a block that forces the user to visualize the slide as it will appear on the front end would remove the need for the character and word count limit plugins that currently are necessary, and provide more of a WYSIWYG experience.

The UnPublish date selector doesn't appear in the sidebar of the new editor, so that, more than the tweaks to enforce a default day, would be where the improvements need to be made on the Unpublish functionality.

**Comment by r-a-y** [comment]
I've made some progress on merging the two plugins.

I don't have anything to show for it yet, but I just wanted some UX feedback regarding the Reveal JS Presentations vs. Slides plugins.

When inheriting the Slides Block Editor interface, do we want all the different sidebar block options? Here's a screenshot that you shared with me in an email:
![image](https://user-images.githubusercontent.com/505921/87606268-14b0cf00-c6ea-11ea-8511-b7961b8410e9.png)

Reveal JS Presentations has some slideshow options, but they are under the `'presentation'` taxonomy page.

Which options will have precedence over the other? I'm basically rewriting the Reveal JS Presentations plugin, so we can do whatever we want.

**Comment by mrjarbenne** [comment]
Of the slide specific options available:
- the Background image is utilized currently, in some situations only in order to create a full sized image on a slide, as the current plugin doesn't handle inline images well.
- the video/iframe functionality has been problematic in the past because the most popular use-case for this is on a display in a hallway, set to autoplay and autoadvance the slide at specified intervals. There are a couple of Reveal plugins that get around this by detecting the video length and delaying the slide advance until the video is complete. Here's one if it's something that could be implemented (http://hamelot.io/other/autoplaying-vimeo-slideshow-with-reveal-js/) related: https://github.com/hwdsb/reveal-js-presentations/issues/7

Of the Presentation specific options:
- We need the ability to Autoplay the slide show, and loop the slide show
- We need the ability to tweak the duration of each slide before a transition
- There are a number of other options in Under the Edit menu in the Presentation Taxonomy screen.
-- Hiding the address bar was something we tweaked so that full screen mode looked better on a monitor
-- Touch navigation is important if that's something that needs to be specifically turned on

Most of the other options we need are already in the new plugin, under the Presentation Block settings: 

<img width="281" alt="2020-07-23_11-39-16" src="https://user-images.githubusercontent.com/480667/88306796-2ccfb200-ccd9-11ea-8abb-1330e1205665.png">

This is the Taxonomy settings page from the old plugin. The Autoplay and loop settings are the only ones I think are really important, but let me know if there are other settings on that page you have questions about: 

<img width="1076" alt="2020-07-23_11-26-23" src="https://user-images.githubusercontent.com/480667/88305742-da41c600-ccd7-11ea-92e7-084a2bb74b9d.png">

**Comment by r-a-y** [comment]
For the block settings, I think the Setup, Controls and PDF options should be removed. These are all settings that should be at the presentation level and not the slide. There are probably a few others that can be removed as well, but those are the ones that immediately stood out to me.

I've made some good progress. I will put up a test version on the boreal site in the next day or so and we can iterate from there.

Here's a screenshot of the Presentations taxonomy page:

![Screenshot of Presentations taxonomy page](https://user-images.githubusercontent.com/505921/88587538-9a673f80-d045-11ea-96ec-ad640e5aec91.png)

I will probably add some more options from Reveal as it makes sense to: https://revealjs.com/config/
